mountpoint-s3/src/prefetch.rs

```diff
@@ -176,6 +176,38 @@ impl Default for PrefetcherConfig {
     }
 }
 
+/// Provide the maximum read size for the prefetcher, for which there is one prefetcher per file handle.
+///
+/// This allows a way to override the prefetch window rather than using the hardcoded default within Mountpoint.
+/// We do not recommend using the override, and it may be removed at any time.
+///
+/// This unstable override is expected to be removed once adaptive prefetching based on available memory is available:
+/// https://github.com/awslabs/mountpoint-s3/issues/987
+fn determine_max_read_size() -> usize {
+    let env_var_key = "UNSTABLE_MOUNTPOINT_MAX_PREFETCH_WINDOW_SIZE";
+    const DEFAULT_READ_WINDOW_SIZE: usize = 2 * 1024 * 1024 * 1024;
+
+    match std::env::var_os(env_var_key) {
+        Some(val) => match val.to_string_lossy().parse() {
+            Ok(val) => {
```

Comment:
Do we need to set a lower bound? Have you checked what happens with 0? Or with a value smaller than part size or request size?

Reply:
Checking with 0, it does appear to continue progressing. As discussed, it's likely that the read window begins at 1MiB+128KiB and then increments by this amount because any scale up would always be bigger than 0.